### PR TITLE
Cleanup in case of failures in PubSubAdmin constructor

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes blocker SonarCloud finding:
"Use try-with-resources or close this "TopicAdminClient" in a "finally" clause."

### CODE REVIEW

**Review by elefeint — DISMISSED** [review]
Wait, so this will result in the admin clients getting closed immediately after getting created. I think Sonar might be wrong on this one.

We do want to close the client library objects in maybe `@PreDestroy`.

**Comment by meltsufin** [comment]
> Wait, so this will result in the admin clients getting closed immediately after getting created. I think Sonar might be wrong on this one.
> 
> We do want to close the client library objects in maybe `@PreDestroy`.

Good catch! And the integration tests caught it too!

**Comment by meltsufin** [comment]
> We do want to close the client library objects in maybe `@PreDestroy`.

The class already implements `AutoCloseable`. So, Spring will try to call the `close()` method already.

File: spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java [hunk 1]
```diff
@@ -54,11 +54,11 @@
 
 	private static final String NO_TOPIC_SPECIFIED = "No topic name was specified.";
 
-	private final String projectId;
+	private String projectId;
 
-	private final TopicAdminClient topicAdminClient;
+	private TopicAdminClient topicAdminClient;
```
Comment (ttomsu) [inline spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java:59]: IMO keeping these `final` is a greater benefit than the small amount of duplicate `Assert.notNull` code in the constructors, which, if I estimate correctly, may just be ensuring the project ID provider is not null and the project ID it provides is not null.

Comment (meltsufin) [reply spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java]: Agreed.

File: spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java [hunk 2]
```diff
@@ -68,19 +68,35 @@
...
-				TopicAdminClient.create(
-						TopicAdminSettings.newBuilder()
-								.setCredentialsProvider(credentialsProvider)
-								.build()),
-				SubscriptionAdminClient.create(
-						SubscriptionAdminSettings.newBuilder()
-						.setCredentialsProvider(credentialsProvider)
-						.build()));
+			CredentialsProvider credentialsProvider) throws PubSubException {
+		SubscriptionAdminClient subscriptionClient;
```
Comment (ttomsu) [inline spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java:75]:
```suggestion
```

**Review by ttomsu — APPROVED** [review]

**Pull request merged**